Java/CoreCrypto/src/main/java/com/breadwallet/corecrypto/Blockchains.java

```diff
@@ -250,19 +206,41 @@
     );
 
     private static Currency CURRENCY_BRD_ROPSTEN = new Currency(
-            "ethereum-ropsten" + ADDRESS_BRD_TESTNET,
+            makeCurrencyIdentifierErc20("ethereum-ropsten:", ADDRESS_BRD_TESTNET),
             "BRD Token Testnet",
-            "BRD",
+            "brd",
             "erc20",
             "ethereum-ropsten",
             ADDRESS_BRD_TESTNET,
             true,
-            ImmutableList.of(
-                    BRD_INT,
-                    BRD_BRD
-            )
+            makeCurrencyDemominationsErc20("brd", UnsignedInteger.valueOf(18))
     );
 
+    private static String makeCurrencyIdentifierErc20(String blockchainId, String address) {
+        return String.format(Locale.ROOT, "%s:%s", blockchainId, address);
+    }
+
+    /* package */
+    static List<CurrencyDenomination> makeCurrencyDemominationsErc20 (String code, UnsignedInteger decimals) {
+        String name = code.toUpperCase(Locale.ROOT);
+        code = code.toLowerCase(Locale.ROOT);
+
+        return ImmutableList.of(
+                new CurrencyDenomination(
+                        String.format(Locale.ROOT, "%s Token INT", name),
+                        String.format(Locale.ROOT, "%si", code),
```

Comment:
I don't think these are always ERC20 tokens (although in practice I think they are).

Reply:
I don't follow the comment here; isn't this an exact replica of `makeCurrencyDemominationsERC20` that you just added in Swift?